components/todo/todoItem.js

```diff
@@ -1,83 +1,107 @@
-      color: gray;
-      text-decoration: line-through;
-      background: #CCDAE0;
-    }
-    .todoName {
-      cursor: pointer;
-    }
-    
-    #deleteTodo{
-      display: inline-block;
-    }
+import { LitElement, html, css } from 'lit-element';
 
-  </style>
-  <div class="todoItem">
-  <p class="todoName"></p>
-  <custom-button id="deleteTodo"> - </custom-button>
-  </div>
-`;
+/**
+ * Class for each item in Todo.
+ *
+ * @class TodoItem
+ * @extends {LitElement}
+ */
+class TodoItem extends LitElement {
+  /**
+   * Returns the style for the component.
+   *
+   * @readonly
+   * @static
+   * @memberof TodoItem
+   */
+  static get styles() {
+    return css`
+      .todoItem {
+        display: flex;
+        align-items: center;
+        margin: 8px 0px;
+        padding: 4px 8px;
+        border: 1px solid #99b1bc;
+        border-radius: 8px;
+      }
+      p {
+        flex: 3;
+      }
+      .done {
+        color: gray;
+        text-decoration: line-through;
+        background: #ccdae0;
+      }
+      .todoName {
+        cursor: pointer;
+      }
 
-class TodoItem extends HTMLElement {
-  static get observedAttributes() {
-    return ['complete'];
+      #deleteTodo {
+        display: inline-block;
+      }
+    `;
   }
 
-  constructor() {
-    super();
-
-    this.attachShadow({ mode: 'open' });
-    this.shadowRoot.appendChild(
-      document.importNode(todoItemTemplate.content, true)
-    );
-
-    this.todoName = this.shadowRoot.querySelector('.todoName');
-    this.todoName.innerText = this.getAttribute('name');
-
-    this.deleteTodoBtn = this.shadowRoot.querySelector('#deleteTodo');
+  /**
+   * Declares properties for component.
+   *
+   * @readonly
+   * @static
+   * @memberof TodoItem
+   */
+  static get properties() {
+    return {
+      id: { type: String },
+      name: { type: String },
+      complete: { type: String },
+      actions: { type: Object },
+    };
   }
 
-  attributeChangedCallback(attrName, oldVal, newVal) {
-    if (attrName === 'complete') {
-      if (newVal === 'true') {
-        this.shadowRoot.querySelector('.todoItem').classList.add('done');
-      } else {
-        this.shadowRoot.querySelector('.todoItem').classList.remove('done');
-      }
-    }
-  }
+  /**
+   * Creates an instance of TodoItem.
+   * @memberof TodoItem
+   */
+  constructor() {
+    super();
 
-  connectedCallback() {
-    this.todoName.addEventListener('click', this.dispatchToggleEvent);
-    this.deleteTodoBtn.addEventListener('click', this.dispatchDeleteEvent);
+    this.id = `${+new Date()}`;
+    this.name = 'Untitiled';
+    this.complete = 'false';
   }
 
-  disconnectedCallback() {
-    this.todoName.addEventListener('click', this.dispatchToggleEvent);
-    this.deleteTodoBtn.removeEventListener('click', this.dispatchDeleteEvent);
+  /**
+   *
+   *
+   * @returns
+   * @memberof TodoItem
+   */
+  render() {
+    return html`
+      <div class="todoItem ${this.complete === 'true' ? 'done' : ''}">
```

Comment:
Instead of doing `this.complete === 'true'`
we can simply do, `${this.complete ? 'done' :' '`

or better use short circuit like,` this.complete && 'done'`. 

something like that.